impeller/entity/contents/solid_stroke_contents.cc

```diff
@@ -29,395 +28,56 @@ const Color& SolidStrokeContents::GetColor() const {
-      vtx_builder.AppendVertex(vtx);
-      vtx_builder.AppendVertex(vtx);
-
-      vtx.position = polyline.points[contour_start_point_i];
-      // Append two vertices at the beginning of the new contour, which appends
-      // two triangles of zero area.
-      vtx_builder.AppendVertex(vtx);
-      vtx_builder.AppendVertex(vtx);
-    }
-
-    // Generate start cap.
-    if (!polyline.contours[contour_i].is_closed) {
-      cap_proc(vtx_builder, polyline.points[contour_start_point_i], -offset,
-               smoothing);
-    }
-
-    // Generate contour geometry.
-    for (size_t point_i = contour_start_point_i; point_i < contour_end_point_i;
-         point_i++) {
-      if (point_i > contour_start_point_i) {
-        // Generate line rect.
-        vtx.position = polyline.points[point_i - 1] + offset;
-        vtx_builder.AppendVertex(vtx);
-        vtx.position = polyline.points[point_i - 1] - offset;
-        vtx_builder.AppendVertex(vtx);
-        vtx.position = polyline.points[point_i] + offset;
-        vtx_builder.AppendVertex(vtx);
-        vtx.position = polyline.points[point_i] - offset;
-        vtx_builder.AppendVertex(vtx);
-
-        if (point_i < contour_end_point_i - 1) {
-          compute_offset(point_i + 1);
-
-          // Generate join from the current line to the next line.
-          join_proc(vtx_builder, polyline.points[point_i], previous_offset,
-                    offset, miter_limit, smoothing);
-        }
-      }
-    }
-
-    // Generate end cap or join.
-    if (!polyline.contours[contour_i].is_closed) {
-      cap_proc(vtx_builder, polyline.points[contour_end_point_i - 1], offset,
-               smoothing);
-    } else {
-      join_proc(vtx_builder, polyline.points[contour_start_point_i], offset,
-                contour_first_offset, miter_limit, smoothing);
-    }
-  }
-
-  return vtx_builder.CreateVertexBuffer(buffer);
+  return geometry_->GetCoverage(entity.GetTransformation());
 }
 
 bool SolidStrokeContents::Render(const ContentContext& renderer,
                                  const Entity& entity,
                                  RenderPass& pass) const {
-  if (stroke_size_ < 0.0) {
-    return true;
-  }
-
+  using VS = SolidFillVertexShader;
+  using FS = SolidFillFragmentShader;
   VS::VertInfo vert_info;
   vert_info.mvp = Matrix::MakeOrthographic(pass.GetRenderTargetSize()) *
                   entity.GetTransformation();
-  Scalar determinant = entity.GetTransformation().GetDeterminant();
-  if (determinant == 0) {
-    return true;
-  }
 
   FS::FragInfo frag_info;
   frag_info.color = color_.Premultiply();
 
   Command cmd;
   cmd.primitive_type = PrimitiveType::kTriangleStrip;
   cmd.label = "Solid Stroke";
+  cmd.stencil_reference = entity.GetStencilDepth();
+
+  auto geometry_result = geometry_->GetPositionBuffer(renderer, entity, pass);
+
   auto options = OptionsFromPassAndEntity(pass, entity);
-  if (!color_.IsOpaque()) {
+  if (geometry_result.prevent_overdraw) {
```

Comment:
I think this is the right check :)

Once we merge solid fill + stroke, we'll need to retain these clips/options I'm assuming.